plugins/inputs/prometheus_scraper/metrics_filter_test.go

```diff
@@ -30,54 +30,119 @@ func buildTestData(pass int, drop int) (result PrometheusMetricBatch) {
 
-	batch := buildTestData(2, 2)
-	batch = p.Filter(batch)
 
-	assert.Equal(t, 3, p.maxDropMetricsLogged)
-	assert.Equal(t, 2, len(batch))
-	assert.Equal(t, false, p.hitMaxLimit)
-	assert.Equal(t, 2, len(p.droppedMetrics))
+	assert.Equal(t, 3, metricFilterWithoutDroppingUnsupportedMetric.maxDropMetricsLogged)
+	assert.Equal(t, 4, len(metricBatchWithoutDroppingUnsupportedMetric))
+	assert.Equal(t, false, metricFilterWithoutDroppingUnsupportedMetric.hitMaxLimit)
+	assert.Equal(t, 0, len(metricFilterWithoutDroppingUnsupportedMetric.droppedMetrics))
 }
 
 func TestMetricsFilterFilter_HitMaxDroppedMetrics(t *testing.T) {
+	var dropUnsupportedMetric bool
+	//Filter Metric with DroppingUnsupportedMetric equals to true
+	metricFilterWithDroppingUnsupportedMetric := &MetricsFilter{maxDropMetricsLogged: 3}
 
-	p := &MetricsFilter{maxDropMetricsLogged: 3}
+	dropUnsupportedMetric = true
+	metricBatchWithDroppingUnsupportedMetric := buildTestData(7, 3)
+	metricBatchWithDroppingUnsupportedMetric = metricFilterWithDroppingUnsupportedMetric.Filter(metricBatchWithDroppingUnsupportedMetric,dropUnsupportedMetric)
 
-	batch := buildTestData(7, 3)
-	batch = p.Filter(batch)
 
-	assert.Equal(t, 3, p.maxDropMetricsLogged)
-	assert.Equal(t, 7, len(batch))
-	assert.Equal(t, true, p.hitMaxLimit)
-	assert.Equal(t, 0, len(p.droppedMetrics))
+	assert.Equal(t, 3, metricFilterWithDroppingUnsupportedMetric.maxDropMetricsLogged)
+	assert.Equal(t, 7, len(metricBatchWithDroppingUnsupportedMetric))
+	assert.Equal(t, true, metricFilterWithDroppingUnsupportedMetric.hitMaxLimit)
+	assert.Equal(t, 0, len(metricFilterWithDroppingUnsupportedMetric.droppedMetrics))
+
+	//Filter Metric with DroppingUnsupportedMetric equals to false
+	metricFilterWithoutDroppingUnsupportedMetric := &MetricsFilter{maxDropMetricsLogged: 3}
+
+	dropUnsupportedMetric = false
+	metricBatchWithoutDroppingUnsupportedMetric := buildTestData(7, 3)
+	metricBatchWithoutDroppingUnsupportedMetric = metricFilterWithoutDroppingUnsupportedMetric.Filter(metricBatchWithoutDroppingUnsupportedMetric,dropUnsupportedMetric)
+
+
+	assert.Equal(t, 3, metricFilterWithoutDroppingUnsupportedMetric.maxDropMetricsLogged)
+	assert.Equal(t, 10, len(metricBatchWithoutDroppingUnsupportedMetric))
+	assert.Equal(t, false, metricFilterWithoutDroppingUnsupportedMetric.hitMaxLimit)
+	assert.Equal(t, 0, len(metricFilterWithoutDroppingUnsupportedMetric.droppedMetrics))
 }
 
 func TestMetricsFilterFilter_ExceedMaxDroppedMetrics(t *testing.T) {
+	var dropUnsupportedMetric bool
+	//Filter Metric with DroppingUnsupportedMetric equals to true
+	metricFilterWithDroppingUnsupportedMetric := &MetricsFilter{maxDropMetricsLogged: 3}
+
+	dropUnsupportedMetric = true
+	metricBatchWithDroppingUnsupportedMetric := buildTestData(2, 12)
+	metricBatchWithDroppingUnsupportedMetric = metricFilterWithDroppingUnsupportedMetric.Filter(metricBatchWithDroppingUnsupportedMetric,dropUnsupportedMetric)
```

Comment:
This bit of code looks like it is duplicated.

Two things:
1. What is the purpose of the last two lines? We build the test data and then always filter out unsupported metrics? 
2. Can all of this logic be put inside of the `buildTestData()` helper function? 
3.

Reply:
1. Technically yes, we build two types of metrics: counter which supposed to be pass and histogram, which supposed to be dropped.
2. Great idea though. WIll try to do that.